tests/executables_src/testSubdeviceBackend.cpp

```diff
@@ -586,6 +586,63 @@ BOOST_AUTO_TEST_CASE(test3regsByteOffset1) {
 
 /*********************************************************************************************************************/
 
+BOOST_AUTO_TEST_CASE(testAreaHandshake1) {
+  setDMapFilePath("subdeviceTestAreaHandshake.dmap");
+
+  Device dev;
+  dev.open("SUBDEV4");
+  Device target;
+  target.open("TARGET1");
+
+  auto acc1 = dev.getScalarRegisterAccessor<double>("APP.0.MY_REGISTER1");
+  auto acc2 = dev.getScalarRegisterAccessor<double>("APP.0.MY_REGISTER2");
+  auto acc3 = dev.getOneDRegisterAccessor<int>("APP.0.MY_AREA1", 6, 0);
+  auto accArea = target.getOneDRegisterAccessor<int32_t>("APP.0.THE_AREA", 10, 0, {AccessMode::raw});
+  target.getScalarRegisterAccessor<int32_t>("APP.0.THE_AREA");
+  auto accS = target.getScalarRegisterAccessor<int32_t>("APP.1.STATUS");
+  std::atomic<bool> done;
+  std::thread t;
+
+  BOOST_CHECK_THROW(acc1.read(), ChimeraTK::logic_error);
+  std::vector<int> vec = {1, 2, 3, 4, 5, 6};
+
+  accS = 1;
+  accS.write();
+  done = false;
+  t = std::thread([&] {
+    acc1 = 1897;
+    acc2 = 1897;
+    acc3 = vec;
+    acc1.write();
+    acc2.write();
+    acc3.write();
+    done = true;
+  });
+  usleep(10000);
+  BOOST_CHECK(done == false);
+  int countStatusResets = 0;
+  // the dummyForAreaHandshake backend which we use for this test does not set back the status register. we do it manually from the test, and count how often we need do so
+  // Like this we can check that the accessor waits on status==0 _each_ time before writing,
+  // in particular each array entry counts.
+  while(!done) {
+    countStatusResets++;
+    accS = 0;
+    accS.write();
+    usleep(200000);
+  }
+  BOOST_CHECK(countStatusResets >= 8); // do not strictly compare with 8 since we have little control over timing
```

Comment:
Why no exact comparison? The accessor should always write exactly all elements once. How can it become bigger than 8? The test is writing in total 8 elements exactly. All writes are complete before this comparison is done (this is  synchronised via `done`). This seems to be somewhat fishy...

Reply:
changed to waiting for status=busy in loop